docs/4-functional-programming/fp-hometask/src/dataConverter.ts

```diff
@@ -0,0 +1,35 @@
+import { Account, Image, User } from 'types';
+import { Row } from './components';
+
+export const dataConverter = (
+  users: User[],
+  accounts: Account[],
+  images: Image[]
+): Row[] => {
+  return users.map(({ userID, username, country, name }) => {
+    let avatar = 'fallback-image';
+    let lastPayments = 0;
+    let posts = 0;
+
+    const image = images.find(image => image.userID === userID);
+    const account = accounts.find(acc => acc.userID === userID);
+
+    if (image) {
+      avatar = image.url;
+    }
+
+    if (account) {
+      lastPayments = account?.payments?.[0]?.totalSum ?? 0;
+      posts = account?.posts ?? 0;
+    }
+
+    return {
+      avatar,
+      username,
+      country,
+      name,
+      lastPayments,
+      posts,
+    };
+  });
+};
```

Comment:
try Functional Programming instead of conventional way. For example:


```
    type DataWithUserId = { userID: string; }

    const addAvatar: <T extends DataWithUserId>(data: T) => T & { avatar: string };
    // TBD

    const addAccount: <T extends DataWithUserId>(data: T) => T & { posts: number, lastPayments: number };
    // TBD

    const transformDataToRow: (data: User & { avatar: string, posts: number, lastPayments: number }) => Row;
    // TBD

    return users.map((user: User) => transformDataToRow(addAvatar(addAccount(user))));
    // OR
    // return users.map(compose(addAvatar, addAccount, transformDataToRow));
```

Reply:
it seems to me I understood your suggestion and implemented correctly

Reply:
you are correct